tests/context_tests.rs

```diff
@@ -1853,6 +1859,60 @@ mod test_ctx_load {
     }
 }
 
+mod test_evict_control {
+    use super::*;
+
+    #[test]
+    fn test_basic_evict_control() {
+        let mut context = create_ctx_without_session();
+
+        let auth = Auth::default();
+        let owner_object_handle = ObjectHandle::from(Hierarchy::Owner.esys_rh());
+        context
+            .tr_set_auth(owner_object_handle, &auth)
+            .expect("Failed to set password for owner to empty password.");
+
+        context.set_sessions((Some(Session::Password), None, None));
+
+        let auth_value_primary = Auth::try_from(vec![1, 2, 3, 4, 5])
+            .expect("Failed to crate auth value for primary key");
+        let prim_key_handle = context
+            .create_primary_key(
+                Hierarchy::Owner,
+                &decryption_key_pub(),
+                Some(auth_value_primary).as_ref(),
+                None,
+                None,
+                &[],
+            )
+            .expect("Failed to create primary key");
+
+        let persistent_tpm_handle =
+            PersistentTpmHandle::new(u32::from_be_bytes([0x81, 0x0, 0x0, 0x1]))
+                .expect("Failed to create persitent tpm handle");
+        let persistent = Persistent::Persistent(persistent_tpm_handle);
+        let mut persistent_handle = context
+            .evict_control(Provision::Owner, prim_key_handle.into(), persistent)
+            .expect("Failed to make the primary key handle persistent");
+
+        context
+            .tr_close(&mut ObjectHandle::from(prim_key_handle))
+            .expect("Failed to close primary key handle");
+        context
+            .tr_close(&mut persistent_handle)
+            .expect("Failed to close persistant handle");
+
+        let retireved_persistant_handle = context
+            .tr_from_tpm_public(TpmHandle::Persistent(persistent_tpm_handle))
+            .expect("Failed to load the persistant handle");
+
+        assert_ne!(
+            retireved_persistant_handle,
+            ObjectHandle::from(ESYS_TR_NONE)
+        );
```

Comment:
Could you also add a step to evict the persistent object and then try to load the TPM resource again?